src/sql/parts/commandLine/common/commandLineService.ts

```diff
@@ -0,0 +1,80 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+import { ConnectionProfile } from 'sql/parts/connection/common/connectionProfile';
+import { ICommandLineProcessing } from 'sql/parts/commandLine/common/commandLine';
+import { IConnectionManagementService } from 'sql/parts/connection/common/connectionManagement';
+import { ICapabilitiesService } from 'sql/services/capabilities/capabilitiesService';
+import { IEnvironmentService } from 'vs/platform/environment/common/environment';
+import * as Constants from 'sql/parts/connection/common/constants';
+import { IQueryEditorService } from 'sql/parts/query/common/queryEditorService';
+import * as platform from 'vs/platform/registry/common/platform';
+import { ConnectionProviderProperties, IConnectionProviderRegistry, Extensions as ConnectionProviderExtensions } from 'sql/workbench/parts/connection/common/connectionProviderExtension';
+import * as TaskUtilities from 'sql/workbench/common/taskUtilities';
+import { IObjectExplorerService } from 'sql/parts/objectExplorer/common/objectExplorerService';
+import { IEditorService } from 'vs/workbench/services/editor/common/editorService';
+
+export class CommandLineService implements ICommandLineProcessing {
+	private _connectionProfile : ConnectionProfile;
+	private _showConnectionDialog: boolean;
+
+	constructor(
+		@IConnectionManagementService private _connectionManagementService : IConnectionManagementService,
+		@ICapabilitiesService private _capabilitiesService : ICapabilitiesService,
+		@IEnvironmentService private _environmentService : IEnvironmentService,
+		@IQueryEditorService private _queryEditorService : IQueryEditorService,
+		@IObjectExplorerService private _objectExplorerService : IObjectExplorerService,
+		@IEditorService private _editorService : IEditorService,
+	)
+	{
+		let profile = null;
+		if (this._environmentService && this._environmentService.args.server)
```

Comment:
_environmentService.args will never be undefined?

Reply:
the _args parameter to the EnvironmentService constructor is not optional and the EnvironmentService itself never checks it for undefined/null.